paramiko/transport.py

```diff
@@ -1021,10 +1021,16 @@ def global_request(self, kind, data=None, wait=True):
         :param bool wait:
             ``True`` if this method should not return until a response is
             received; ``False`` otherwise.
+        :param float timeout:
+            optional timeout making a global request, no default set
```

Comment:
I realise this is somewhat modelled on `open_channel()`, but having both `wait=` and `timeout=` is weird. Timeout is a superset of the wait behaviour, but means nothing without it.